convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,219 @@
+    id = "EFIBOOTMGR_UTILITY_INSTALLED"
+    dependencies = ("NEW_DEFAULT_EFI_BIN",)
+
+    def run(self):
+        """Check if the Efibootmgr utility is installed"""
+        super(EfibootmgrUtilityInstalled, self).run()
+
+        if not os.path.exists("/usr/sbin/efibootmgr"):
+            self.set_result(
+                level="ERROR",
+                id="EFIBOOTMGR_UTILITY_NOT_INSTALLED",
+                title="Efibootmgr utility is not installed",
+                description="The /usr/sbin/efibootmgr utility is not installed.",
+                remediations="Install the efibootmgr utility via YUM/DNF.",
+            )
+
+
+class CopyGrubFiles(actions.Action):
+    id = "COPY_GRUB_FILES"
+    dependencies = ("EFIBOOTMGR_UTILITY_INSTALLED",)
+
+    def run(self):
+        """Copy grub files from centos/ dir to the /boot/efi/EFI/redhat/ dir.
+
+        The grub.cfg, grubenv, ... files are not present in the redhat/ directory
+        after the conversion on a CentOS Linux system. These files are usually created
+        during the OS installation by anaconda and have to be present in the
+        redhat/ directory after the conversion.
+
+        The copy of the centos/ directory should be ok. In case of the conversion
+        from Oracle Linux, the redhat/ directory is already used.
+        """
+        super(CopyGrubFiles, self).run()
+
+        if systeminfo.system_info.id != "centos":
+            logger.debug("Did not perform copying of GRUB files - only related to CentOS Linux.")
+            return
+
+        # TODO(pstodulk): check behaviour for efibin from a different dir or with a different name for the possibility of
+        #  the different grub content...
+        # E.g. if the efibin is located in a different directory, are these two files valid?
+        logger.info("Copying GRUB2 configuration files to the new UEFI directory %s." % RHEL_EFIDIR_CANONICAL_PATH)
+        src_files = [
+            os.path.join(CENTOS_EFIDIR_CANONICAL_PATH, filename) for filename in ["grubenv", "grub.cfg", "user.cfg"]
+        ]
+        required = src_files[:2]
+
+        # If at least one file exists, this will be skipped. Otherwise, if all
+        # are missing, this will be a hit.
+        if not any(os.path.exists(filename) for filename in src_files):
+            # Get a list of files that are missing that are required and does
+            # not exist.
+            missing_files = [
+                filename for filename in src_files if filename in required and not os.path.exists(filename)
+            ]
+            # without the required files user should not reboot the system
+            self.set_result(
+                level="ERROR",
+                id="UNABLE_TO_FIND_REQUIRED_FILE_FOR_GRUB_CONFIG",
+                title="Unable to find required file for GRUB config",
+                description="Unable to find the original file required for GRUB configuration at: %s"
+                % ", ".join(missing_files),
```

Comment:
I'm not entirely sure atm what we do with the copying of grub configs bit something along these lines is more friendly to the user:
```suggestion
                title="Couldn't find system GRUB config",
                description="Couldn't find any GRUB config files in the current system which is required for configuring UEFI for RHEL:{}".format("\n- ".join(missing_files)),
```